src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -0,0 +1,109 @@
+/* global Promise */
+import { _, $, loc } from 'okta';
+import BaseView from '../internals/BaseView';
+import BaseForm from '../internals/BaseForm';
+import Logger from '../../../util/Logger';
+
+const request = (url, method = 'GET', body = null, timeout = 1000) => {
+  return $.ajax({
+    url,
+    method,
+    timeout,
+    contentType: 'application/json',
+    data: body,
+  });
+};
+
+const Body = BaseForm.extend({
+  noButtonBar: true,
+
+  className: 'ion-form device-challenge-poll',
+
+  title: loc('signin', 'login'),
+
+  initialize () {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    this.deviceChallengePollRemediation = this.options.appState.getCurrentViewState();
+    this.doChallenge();
+    this.startPolling();
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    this.stopPolling();
+  },
+
+  doChallenge () {
+    const deviceChallenge = this.options.appState.get('currentState')[
+      this.deviceChallengePollRemediation.relatesTo
+    ];
+    switch (deviceChallenge.challengeMethod) {
+    case 'LOOPBACK':
+      this.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+      break;
+    case 'custom-scheme':
+      this.doCustomURI(deviceChallenge.href);
+      break;
+    }
+  },
+
+  doLoopback (authenticatorDomainUrl = '', ports = [], challengeRequest = '') {
+    let currentPort;
+    let foundPort = false;
+
+    const getAuthenticatorUrl = (path) => {
+      return `${authenticatorDomainUrl}:${currentPort}/${path}`;
+    };
+
+    const checkPort = () => {
+      return request(getAuthenticatorUrl('probe'));
+    };
+
+    const onPortFound = () => {
+      foundPort = true;
+      return request(getAuthenticatorUrl('challenge'), 'POST', JSON.stringify({ challengeRequest }) , 3000);
+    };
+
+    const onFailure = () => {};
```

Comment:
What is the use of this empty `onFailure `

Reply:
It's more like a placeholder for now until we know if we need UX to handle it.

Reply:
should we add a comment then?